torch/distributed/fsdp/_optim_utils.py

```diff
@@ -2063,13 +2063,13 @@ def return_fn(fqn_to_param_info):
 
 @no_type_check
 def _set_optim_use_dtensor(
-    module: nn.Module,
+    fsdp_state: _FSDPState,
```

Comment:
I think we also need to update the caller of `_set_optim_use_dtensor` to pass fsdp_state into `_set_optim_use_dtensor` as well.

https://github.com/pytorch/pytorch/blob/main/torch/distributed/fsdp/fully_sharded_data_parallel.py#L767C27-L767C27

Reply:
For a FullyShardedDataParallel module, itself is the fsdp_state.